Address SECDIR review feedback from #136 (#139)
ietf-wg-scitt/draft-ietf-scitt-scrapi: draft-ietf-scitt-scrapi.md

informative:
  RFC8792:

>   RFC9921:

Reviewer: ```suggestion
  RFC9921:
  RFC3161:
```

While denial of service attacks are very hard to defend against completely, and Transparency Services are unlikely to be in the critical path of any safety-liable operation, any attack which could cause the _silent_ failure of Signed Statement registration, for example, should be considered in scope.

The impact of DoS attacks can be detected by a client checking that the Transparency Service has registered any submitted Signed Statement and returned a Receipt.
Since verification of Receipts does not require the involvement of the Transparency Service, a DoS attack cannot cause the silent loss of a registration.
However, this relies on clients actively checking for Receipts and does not prevent the disruption itself.

Clients to Transparency Services SHOULD ensure that Receipts are available for their registered Statements, either on a periodic or needs-must basis, depending on the use case.

> Beyond this, implementers of Transparency Services MUST implement general good practice around defending against network attacks such as flooding, including defenses such as rate limiting.

Reviewer: ```suggestion
Beyond this, implementers of Transparency Services MUST follow general good practice around defending against network attacks such as flooding, including defenses such as rate limiting.
```

Insertion attacks are mitigated by the use of the Issuer signature on the Signed Statement, therefore care must be taken in the protection of Issuer keys and credentials to avoid theft and impersonation.

> Transparency Services MAY also implement additional protections such as anomaly detection in order to mitigate the impact of any breach.

Reviewer: ```suggestion
```

Once a statement is made, it is intended to be immutable and non-repudiable, so making it twice should not lead to any particular issues.
There could be issues at the payload level (for instance, the statement "it is raining" may be true when first submitted but not when replayed), but being payload-agnostic implementations of SCITT services cannot be required to worry about that.

> If the semantic content of the payload are time-dependent and susceptible to replay attacks in this way then timestamps MUST be added to the protected header signed by the Issuer.

Reviewer: @copilot add a reference to RFC9921 as a suggested mechanism to do this.

Once a statement is made, it is intended to be immutable and non-repudiable, so making it twice should not lead to any particular issues.
There could be issues at the payload level (for instance, the statement "it is raining" may be true when first submitted but not when replayed), but being payload-agnostic implementations of SCITT services cannot be required to worry about that.

If the semantic content of the payload are time-dependent and susceptible to replay attacks in this way then timestamps MUST be added to the protected header signed by the Issuer.

> {{RFC9921}} defines COSE header parameters for including RFC 3161 timestamp tokens that MAY be used for this purpose.

Reviewer: ```suggestion
{{RFC9921}} defines COSE header parameters for including {{RFC3161}} timestamp tokens that MAY be used for this purpose.
```
